Update Makefile for LWIP integration

razvand opened the pull request:
This PR makes Musl compatible with LWIP and remove some build warnings.

razvand committed:
Makefile: Make Makefile.uk.musl.network compatible with LWIP

Remove the lack of presence of `posix-driver`. `posix-driver` needs to
be enabled for Musl to work with networking support.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

StefanJum commented on Makefile.uk.musl.network:
```diff
@@ -86,7 +86,7 @@ LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/hstrerror.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/htons.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/if_freenameindex.c
-LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/if_indextoname.c
+# LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/if_indextoname.c
```
Removing this source file fails [here](https://git.musl-libc.org/cgit/musl/tree/src/network/getnameinfo.c#n175) with an undefined reference to `if_indextoname` when building a simple `helloworld` app with no `lwip` added.

```suggestion
ifneq ($(CONFIG_LWIP),y)
LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/if_indextoname.c
endif
```

StefanJum commented on Makefile.uk.musl.network:
```diff
@@ -48,16 +48,20 @@ LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/getsockopt.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/setsockopt.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/connect.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/listen.c
+ifneq ($(CONFIG_LWIP),y)
```
```suggestion
ifneq ($(CONFIG_LIBLWIP),y)
```

StefanJum commented on Makefile.uk.musl.network:
```diff
@@ -48,16 +48,20 @@ LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/getsockopt.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/recvfrom.c
 # LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/recvmmsg.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/recvmsg.c
+ifneq ($(CONFIG_LWIP),y)
```
```suggestion
ifneq ($(CONFIG_LIBLWIP),y)
```

StefanJum commented on Makefile.uk.musl.network:
```diff
@@ -86,7 +90,9 @@ LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/hstrerror.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/htonl.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/htons.c
 LIBMUSL_NETWORK_SRCS-y += $(LIBMUSL)/src/network/if_freenameindex.c
+ifneq ($(CONFIG_LWIP),y)
```
```suggestion
ifneq ($(CONFIG_LIBLWIP),y)
```

razvand committed:
Makefile.uk: Use Musl-build compatible build rule

Use `_XOPEN_SOURCE=700` for `LIBMUSL_HDRS_FLAGS-y`, similar to the
original Musl build rule.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

craciunoiuc commented on patches/0013-Change-some-things-in-pthread_create.c.patch:
```diff
@@ -0,0 +1,89 @@
+From cab8414305bdbb07a9adfa72682d110fd7365da9 Mon Sep 17 00:00:00 2001
+From: Dragos Iulian Argint <dragosargint21@gmail.com>
+Date: Sun, 14 Aug 2022 20:14:33 +0300
+Subject: [PATCH] Change some things in pthread_create.c
```
This probably needs a better, more descriptive, title. Also you don't need to specify the file in the Subject.

craciunoiuc reviewed (commented):
<ignore this, you didn't see anything here>

craciunoiuc commented on patches/0001-WIP-Changes-to-remove-warnings.patch:
```diff
@@ -0,0 +1,430 @@
+--- a/src/complex/cacosf.c
++++ b/src/complex/cacosf.c
+@@ -1,3 +1,4 @@
++#define _GNU_SOURCE
+ #include "libm.h"
```
I see this define in a lot of places.

Can't this maybe be moved to some common header. In `libm.h`, or any other subsequent header maybe.

craciunoiuc replied:
I think there's a way to add the define in the Makefile only to specific files.

This would significantly reduce the size of the PR/patch file.

craciunoiuc reviewed (commented):
Hey

Only one comment from my side. It's related to how the patch file is structured.

craciunoiuc commented:
I'm waiting for @StefanJum to add the tag so I'll add mine too.

StefanJum reviewed (approved):
All good @razvand, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

craciunoiuc reviewed (approved):
All good here. Thanks!

Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
